torch/distributed/_sharding_spec/_internals.py

```diff
@@ -40,25 +16,19 @@ class ShardMetadata(object):
         shard_lengths(List[int]): Lengths indicating the length of each
             dimension for this shard. Should have the same rank as the
             original tensor.
-        placement(Device):
+        placement(:class:`torch.distributed.remote_device._RemoteDevice`):
             Specifies the placement of this shard.
-
-            The placement can be a local device or a remote device specified by one
-            of the following remote formats:
-
-                1. "rank:<rank>/<device>" (ex: "rank:0/cuda:0").
-                2. "<worker_name>/<device>" (ex: "trainer0/cuda:0").
     """
 
     __slots__ = ['shard_offsets', 'shard_lengths', 'placement']
 
     shard_offsets: List[int]
     shard_lengths: List[int]
-    placement: Device
+    placement: _RemoteDevice
 
     def __post_init__(self):
-        if not is_valid_device(self.placement):
-            raise ValueError(f'{self.placement} is not a valid device')
+        if isinstance(self.placement, str):
```

Comment:
so it seems `ShardedMetadata` still accepts a str and we convert to a `_RemoveDevice` if it's the case?

Reply:
Yup, so you can pass in str or RemoteDevice for easy of use.